### PR TITLE
fix: space permission on dashboard create

### PR DESCRIPTION
**Pull request by IrakliJani**
closes: https://github.com/lightdash/lightdash/issues/4809

please read the issue for more context.

basically when we are creating dashboard it was defaulting first available space within the project.
if user A changes first space permission to private
and user B has no access to that space
when user B starts to create a dashboard it will try to default to that first space that userB has no permission to 💥 

this one also fixes UI bug where we were not assigning spaceUuid when we were creating a dashboard from space page

### CODE REVIEW

File: packages/backend/src/database/entities/spaces.ts [hunk 1]
```diff
@@ -70,6 +72,22 @@ export const getSpace = async (
...
+            `${SpaceShareTableName}.space_id`,
+            `${SpaceTableName}.space_id`,
+        )
+        .leftJoin(
+            'users',
+            `${SpaceShareTableName}.user_id`,
+            `${UserTableName}.user_id`,
+        )
+        .where((q) => {
+            q.where(`${UserTableName}.user_uuid`, userUuid).orWhere(
```
Comment (rephus) [inline packages/backend/src/database/entities/spaces.ts:86]: :+1:

**Review by rephus — APPROVED** [review]
Amazing job ! 

I'm not sure if we also need to run   a migration to fix all existing spaces ...

If E2e tests work, I think we can merge

**Comment by IrakliJani** [comment]
@rephus spaces/is_private defaults to false so I think we are good here.

**Pull request merged**